src/main/java/uk/gov/justice/digital/delius/jpa/standard/repository/NsiRepository.java

```diff
@@ -1,14 +1,17 @@
 package uk.gov.justice.digital.delius.jpa.standard.repository;
 
-import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 import uk.gov.justice.digital.delius.jpa.standard.entity.Nsi;
 
+import java.util.List;
+
 public interface NsiRepository extends JpaRepository<Nsi, Long> {
 
     // There is no EVENT_ID on NSI
     @Query("select n from Nsi n where n.event.eventId = :eventId AND n.offenderId = :offenderId")
     List<Nsi> findByEventIdAndOffenderId(@Param("eventId") Long eventId, @Param("offenderId") Long offenderId);
+
+    Nsi getByNsiId(Long nsiId);
```

Comment:
do you need this, or can you use the super.findById() which returns an optional?